### PR TITLE
Add test to ensure emails are validated & remove duplicate validation 

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Extends https://github.com/civicrm/civicrm-core/pull/23415 to also test that emails are validated - tests demonstrate that they are caught in the `isErrorInCoreData` function & the other places (in old code) are redundant

Before
----------------------------------------
Validation of emails occurs in multiple places & we are too scared to change that because of the lack of tests

After
----------------------------------------
Tests added, redundant code removed

Technical Details
----------------------------------------
If https://github.com/civicrm/civicrm-core/pull/23415 is merged first it can be rebased out

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Contact/Import/Parser/Contact.php [hunk 1]
```diff
@@ -2244,24 +2218,6 @@ protected function handleDuplicateError(array $newContact, $statusFieldName, arr
...
-          continue;
-        }
-        if ($email = CRM_Utils_Array::value('email', $values)) {
-          // validate each email
-          if (!CRM_Utils_Rule::email($email)) {
-            throw new CRM_Core_Exception('No valid email address');
-          }
-
-          // check for loc type id.
-          if (empty($values['location_type_id'])) {
```
Comment (eileenmcnaughton) [inline CRM/Contact/Import/Parser/Contact.php:2259]: I don't think it can really get to this point with no location_type_id as that is added during mapping - this is old code that used to be in apiv2.
